File: pow.cpp
double pow(double base,int pow)
{   
    double result=1.0;
    if(pow<1)
    {
        base=1/base;
    }
    while(pow)
    {
        if(pow&1)
        {
            result=result*base;
        }
        base*=base;
        pow/=2;
    }
    return result;
}

File: pow.py
#calculate the power of number.n is base and e is the exponent
def power(n, e):
    if e == 0:
        return 1
    elif e == 1:
        return n
    else:
        return (n*power(n, e-1))
n = 4
p = 2
print(power(n, p))
